doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -105,197 +129,222 @@ simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
+   .. cmd:: Declare Left Step @one_term
 
-      Adds :n:`@term` to the database used by :tacn:`stepl`.
+      Adds :n:`@one_term` to the database used by :tacn:`stepl`.
 
    This tactic is especially useful for parametric setoids which are not accepted
    as regular setoids for :tacn:`rewrite` and :tacn:`setoid_replace` (see
    :ref:`Generalizedrewriting`).
 
-   .. tacv:: stepl @term by @tactic
-
-      This applies :n:`stepl @term` then applies :token:`tactic` to the second goal.
-
-   .. tacv:: stepr @term by @tactic
-      :name: stepr
+   .. tacn:: stepr @one_term {? by @ltac_expr }
 
-      This behaves as :tacn:`stepl` but on the right-hand-side of the binary
-      relation. Lemmas are expected to be of the form
+      This behaves like :tacn:`stepl` but on the right hand side of the binary
+      relation. Lemmas are expected to be in the form
       :g:`forall x y z, R x y -> eq y z -> R x z`.
 
-   .. cmd:: Declare Right Step @term
+   .. cmd:: Declare Right Step @one_term
 
        Adds :n:`@term` to the database used by :tacn:`stepr`.
 
+Rewriting with definitional equality
+------------------------------------
 
-.. tacn:: change @term
-   :name: change
+.. tacn:: change {? @one_term__from {? at @occs_nums } with } @one_term__to {? @occurrences }
 
-   This tactic applies to any goal. It implements the rule ``Conv`` given in
-   :ref:`subtyping-rules`. :g:`change U` replaces the current goal `T`
-   with `U` providing that `U` is well-formed and that `T` and `U` are
-   convertible.
+   Replaces terms with other :term:`convertible` terms.
+   If :n:`@one_term__from` is not specified, then :n:`@one_term__from` replaces the conclusion and/or
+   the specified hypotheses.  If :n:`@one_term__from` is specified, the tactic replaces occurrences
+   of :n:`@one_term__to` within the conclusion and/or the specified hypotheses.
+
+   :n:`{? @one_term__from {? at @occs_nums } with }`
+     Replaces the occurrences of :n:`@one_term__from` specified by :n:`@occs_nums`
+     with :n:`@one_term__to`, provided that the two :n:`@one_term`\s are
+     convertible.  :n:`@one_term__from` may contain pattern variables such as `?x`,
+     whose value which will substituted for `x` in :n:`@one_term__to`, such as in
+     `change (f ?x ?y) with (g (x, y))` or `change (fun x => ?f x) with f`.
+
+   :n:`@occurrences`
+     If `with` is not specified, :n:`@occurrences` must only specify
+     entire hypotheses and/or the goal; it must not include any
+     :n:`at @occs_nums` clauses.
 
    .. exn:: Not convertible.
       :undocumented:
 
-   .. tacv:: change @term with @term’
+   .. exn:: Too few occurrences.
+      :undocumented:
 
-      This replaces the occurrences of :n:`@term` by :n:`@term’` in the current goal.
-      The term :n:`@term` and :n:`@term’` must be convertible.
+   .. tacn:: now_show @one_term
 
-   .. tacv:: change @term at {+ @natural} with @term’
+      This is a synonym for :n:`change @conversion`. It can be used to
```

Comment:
```suggestion
      This is a synonym for :n:`change @one_term`. It can be used to
```

BTW, you should remove `conversion` from the `nitpick_ignore` list. It would have helped to catch this issue. And it's also time to remove `oriented_rewriter` from this list as well.